kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java

```diff
@@ -20,167 +18,191 @@
+    public KuraException(final KuraErrorCode code) {
+        this.code = code;
+    }
+
+    /**
+     * Builds a new {@link KuraException} instance based on the supplied {@link KuraErrorCode}.
+     *
+     * @param code
+     *            the error code
+     * @param arguments
+     *            the arguments
+     */
+    public KuraException(final KuraErrorCode code, final Object... arguments) {
+        this.code = code;
+        this.arguments = arguments;
+    }
+
+    /**
+     * Builds a new {@link KuraException} instance based on the supplied {@link KuraErrorCode}, an optional Throwable
+     * cause, and optional arguments for the associated exception message.
+     *
+     * @param code
+     *            the error code
+     * @param cause
+     *            the cause
+     * @param arguments
+     *            the arguments
+     */
+    public KuraException(final KuraErrorCode code, final Throwable cause, final Object... arguments) {
+        super(cause);
+        this.code = code;
+        this.arguments = arguments;
+    }
+
+    /**
+     * Factory method to build an {@link KuraException} with the {@link KuraErrorCode#INTERNAL_ERROR} code providing
+     * only a message.
+     *
+     * @param message
+     *            the message
+     * @return the kura exception
+     */
+    public static KuraException internalError(final String message) {
+        return new KuraException(KuraErrorCode.INTERNAL_ERROR, null, message);
+    }
+
+    /**
+     * Factory method to build an {@link KuraException} with the {@link KuraErrorCode#INTERNAL_ERROR} code providing a
+     * cause and a message.
+     *
+     * @param cause
+     *            the cause
+     * @return the kura exception
+     */
+    public static KuraException internalError(final Throwable cause) {
+        return new KuraException(KuraErrorCode.INTERNAL_ERROR, cause, "");
+    }
+
+    /**
+     * Factory method to build an {@link KuraException} with the {@link KuraErrorCode#INTERNAL_ERROR} code providing a
+     * cause and a message.
+     *
+     * @param cause
+     *            the cause
+     * @param message
+     *            the message
+     * @return the kura exception
+     */
+    public static KuraException internalError(final Throwable cause, final String message) {
+        return new KuraException(KuraErrorCode.INTERNAL_ERROR, cause, message);
+    }
+
+    /**
+     * Gets the error code.
+     *
+     * @return the error code
+     */
+    public KuraErrorCode getCode() {
+        return this.code;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getLocalizedMessage() {
+        return this.getLocalizedMessage(Locale.getDefault());
+    }
+
+    /**
+     * Gets the localized message.
+     *
+     * @param locale
+     *            the locale
+     * @return the localized message
+     */
+    private String getLocalizedMessage(final Locale locale) {
+        final String pattern = this.getMessagePattern(locale, this.code);
```

Comment:
I don't think we require a `this.` here for method calling. Also not from our code style?!